libraries/chain/controller.cpp

```diff
@@ -2802,6 +2800,20 @@ struct controller_impl {
                }});
 
          if( s == controller::block_status::incomplete ) {
+            fork_db.apply_if<void>([&](auto& forkdb) {
+               const auto& bsp = std::get<std::decay_t<decltype(forkdb.chain_head)>>(cb.bsp);
+
+               uint16_t if_ext_id = instant_finality_extension::extension_id();
+               assert(bsp->header_exts.count(if_ext_id) > 0); // in all instant_finality block headers
+               const auto& if_ext = std::get<instant_finality_extension>(bsp->header_exts.lower_bound(if_ext_id)->second);
+               if (if_ext.qc_claim.is_strong_qc) {
+                  auto claimed = forkdb.search_on_head_branch(if_ext.qc_claim.block_num);
+                  if (claimed) {
+                     set_if_irreversible_block_num(claimed->core.final_on_strong_qc_block_num);
+                  }
```

Comment:
Actually when searching on head, yes. However, this should be searching on bsp branch not head which is not necessarily the same thing here. Even then I think it can be not found if LIB already advanced past the `claimed->core.final_on_strong_qc_block_num`.